packages/public/server/src/module/Renderer/src/Renderer/Renderer.php

```diff
@@ -99,20 +99,24 @@ public function __construct(FeatureFlagsService $featureFlags, $editorRendererUr
      */
     public function render($input)
     {
+        $key = 'renderer/' . hash('sha512', $input);
+
         if ($this->featureFlags->isEnabled('frontend-legacy-content') && $this->getFormatHelper()->isLegacyFormat($input)) {
-            return $this->renderComponentHelper->__invoke('legacy-content', [
-                'input' => $input,
-            ]);
+            return $this->renderComponentService->render(
+                'legacy-content',
+                ['input' => $input],
+                $key
```

Comment:
Unfortunately, this cache handling is still not perfect since we don't know the revision / revision field that is rendered here so we can't have an intelligent cache key. In the future, we probably have a React Component in the frontend that handles the rendering of a whole entity and caches the stuff together.